test/language/statements/async-generator/yield-star-promise-not-unwrapped.js

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2022 André Bargull. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-generator-function-definitions-runtime-semantics-evaluation
+description: >
+  `yield*` in an async generator does not await promises returned by a manually implemented async iterator.
+flags: [async]
+features: [async-iteration]
+---*/
+
+var innerPromise = Promise.resolve("unwrapped value");
+
+var asyncIter = {
+  [Symbol.asyncIterator]() {
+    return this;
+  },
+  next() {
+    return {
+      done: false,
+      value: innerPromise,
+    };
+  },
+  get return() {
+    throw new Test262Error(".return should not be accessed");
+  },
+  get throw() {
+    throw new Test262Error(".throw should not be accessed");
```

Comment:
In cases like this, I prefer to observe execution with "flag" variables that are then asserted at the end of the test. Reason being: the exception handling mechanisms are themselves being validated, so there's a risk that whatever bug causes this statement to be evaluated would also cause the failure not to be reported.

That said, async tests are fundamentally dependent on asynchronous exception handling semantics, so I don't feel this risk is large enough to warrant another round of review. (I'm sharing my perspective anyway in case it's relevant for future tests and in case anyone out there can explain why I should reconsider.)